database/factories/ProductFactory.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace dnj\Invoice\Database\Factories;
+
+use dnj\Currency\Database\Factories\CurrencyFactory;
+use dnj\Currency\Models\Currency;
+use dnj\Invoice\Models\Invoice;
+use dnj\Invoice\Models\Product;
+use dnj\Number\Number;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class ProductFactory extends Factory
+{
+    protected $model = Product::class;
+
+    public function definition()
+    {
+        return [
+            'title' => fake()->sentence(3),
+            'invoice_id' => Invoice::factory(),
+            'currency_id' => Currency::factory(),
+            'price' => Number::fromInt(0),
+            'discount' => Number::fromInt(0),
+            'total_amount' => Number::fromInt(0),
+            'count' => fake()->numberBetween($min = 1, $max = 10),
+            'meta' => null,
+            'distribution_plan' => null,
+            'distribution' => null,
+            'description' => null,
+        ];
+    }
+
+    public function withCurrency(Currency|CurrencyFactory $currency)
+    {
+        return $this->state(fn () => [
+            'currency_id' => $currency,
+        ]);
+    }
+
+    public function withTile(string $title)
+    {
+        return $this->state(fn () => [
+            'title' => $title,
+        ]);
+    }
+
+    public function withDistributionPlan(array $DistributionPlan)
+    {
+        return $this->state(fn () => [
+            'distribution_plan' => $DistributionPlan,
+        ]);
+    }
+
+    public function withDistribution(array $distribution)
+    {
+        return $this->state(fn () => [
+            'distribution' => $distribution,
+        ]);
+    }
+
+    public function withMeta(array $meta)
+    {
+        return $this->state(fn () => [
+            'meta' => $meta,
+        ]);
+    }
+
+    public function withDescription(string $description)
+    {
+        return $this->state(fn () => [
+            'description' => $description,
+        ]);
+    }
+
+    public function withCount(int $count)
+    {
+        return $this->state(fn () => [
+            'count' => $count,
+        ]);
+    }
+
+    public function withInvoice(int $invoiceId)
```

Comment:
Let's accept Invoice|InvoiceFactory instance instead of just id.

Reply:
done